### PR TITLE
Add new file format design markdown

### PR DESCRIPTION
**Pull request by kangpinghuang**

### CODE REVIEW

File: docs/documentation/cn/extending-doris/doris_storage_optimization.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
+# Doris存储文件格式优化 #
```
Comment (imay) [inline docs/documentation/cn/extending-doris/doris_storage_optimization.md]: you should add this document to path "docs/documentation/cn/internal"

File: docs/documentation/cn/extending-doris/doris_storage_optimization.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+文件中的数据按照page的方式进行组织，page是编码和压缩的基本单位。现在的page类型包括以下几种:
+
+### DataPage ###
+
+DataPage分为两种：nullable和non-nullable的data page。
+
+nullable的data page内容包括：
+```
+
+                 +----------------+
```
Comment (imay) [inline docs/documentation/cn/extending-doris/doris_storage_optimization.md]: first rowid and row count is encoded to a Page header ?

File: docs/documentation/cn/extending-doris/doris_storage_optimization.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                 |     data       |
+                 |----------------|
+                 |    checksum    |
+                 +----------------+
+```
+
+non-nullable data page结构如下：
+
+```
+                 |----------------|
```
Comment (imay) [inline docs/documentation/cn/extending-doris/doris_storage_optimization.md]: first rowid is encoded here?

**Review by imay — COMMENTED** [review]
you should add this doc to

File: docs/documentation/cn/extending-doris/doris_storage_optimization.md (hunk 2 above)
Comment (kangpinghuang) [reply docs/documentation/cn/extending-doris/doris_storage_optimization.md]: In this design, the first rowid is encoded in the data. I agree with you that it is better to move first rowid to Page header

File: docs/documentation/cn/extending-doris/doris_storage_optimization.md (hunk 3 above)
Comment (kangpinghuang) [reply docs/documentation/cn/extending-doris/doris_storage_optimization.md]: Yes, It is now encoded here.

File: docs/documentation/cn/extending-doris/doris_storage_optimization.md (hunk 1 above)
Comment (kangpinghuang) [reply docs/documentation/cn/extending-doris/doris_storage_optimization.md]: OK

**Review by imay — APPROVED** [review]

**Pull request merged**